### PR TITLE
Python 3 fixes to make tests succeed

### PR DESCRIPTION
**Pull request by pietrodn**
I made some fixes to the codebase so that the SITL tests succeed when running on Python 3 (now they fail).

I also ran the tests on Python 2 and they didn't break.

### CODE REVIEW

**Comment by pietrodn** [comment]
To make the tests run on Python 3, pymavlink also needs to be patched. I just [submitted](https://github.com/ArduPilot/pymavlink/pull/138) the patch on the pymavlink repo.

File: dronekit/__init__.py [hunk 1]
```diff
@@ -342,7 +342,7 @@ def release_type(self):
         if self.release is None:
             return None
         types = [ "dev", "alpha", "beta", "rc" ]
-        return types[self.release/64]
+        return types[self.release//64]
```
Comment (peterbarker) [inline dronekit/__init__.py]: This really should be a shift.

File: dronekit/__init__.py [hunk 2]
```diff
@@ -2658,8 +2658,7 @@ def thr_min_callback(self, attr_name, value):
         :param args: The callback to invoke when a change in the parameter is detected.
 
         """
-        attr_name = attr_name.upper()
-        return super(Parameters, self).add_attribute_listener(attr_name, *args, **kwargs)
+        return super(Parameters, self).add_attribute_listener(attr_name.upper(), *args, **kwargs)
```
Comment (peterbarker) [inline dronekit/__init__.py]: Mmmm.... this doesn't look like python3 support to me.

File: dronekit/test/sitl/test_mavlink.py [hunk 3]
```diff
@@ -24,5 +24,6 @@ def callback(*args):
     i = 20
     while not result['success'] and i > 0:
         time.sleep(1)
+        i -= 1
```
Comment (peterbarker) [inline dronekit/test/sitl/test_mavlink.py]: Yowsers.

Not a python3 thing, 'though :-)

File: dronekit/__init__.py [hunk 4]
```diff
@@ -1316,7 +1316,7 @@ def listener(self, name, msg):
                         self._params_last = monotonic.monotonic()
                         self._params_duration = start_duration
                     self._params_set[msg.param_index] = msg
-                self._params_map[msg.param_id] = msg.param_value
+                self._params_map[msg.param_id.rstrip(b'\x00').decode('utf8')] = msg.param_value
```
Comment (peterbarker) [inline dronekit/__init__.py]: I wonder if we should be stripping these at the pymavlink level.

File: dronekit/__init__.py (hunk 2 above)
Comment (pietrodn) [reply dronekit/__init__.py]: Uhm, I think I changed this to print the parameter name while debugging and I forgot to change it back. Should be reverted (will do it).

File: dronekit/__init__.py (hunk 4 above)
Comment (pietrodn) [reply dronekit/__init__.py]: Probably yes. I will look into it.

**Comment by pietrodn** [comment]
Superseded by #788 and #789. Closing.

**Pull request closed without merging**